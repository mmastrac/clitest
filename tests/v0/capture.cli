#!/usr/bin/env clitest --v0

# All grok captures of "word" must match
$ echo "Hello, world!\nGoodbye, world!"
# Expect a grok capture as an environment variable
%SET CAPTURED_WORD ${word}
! Hello, %{WORD:word}!
! Goodbye, %{WORD:word}!

$ echo "Hello, world!"
%EXPECT word "$CAPTURED_WORD"
! Hello, %{WORD:word}!

$ echo "Hello, Earth!"
%EXPECT word "$CAPTURED_WORD"
%EXPECT_FAILURE
! Hello, %{WORD:word}!

$ echo "Hello, $CAPTURED_WORD!"; echo "Goodbye, Earth!"
# Import an environment variable as a grok capture requirement
%EXPECT word "$CAPTURED_WORD"
%EXPECT_FAILURE
! Hello, %{WORD:word}!
! Goodbye, %{WORD:word}!
