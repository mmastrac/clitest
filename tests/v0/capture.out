Running <root>/capture.cli ...

echo "Hello, world!\nGoodbye, world!"
---
Hello, world!
Goodbye, world!
---
✅ OK

echo "Hello, world!"
---
Hello, world!
---
✅ OK

echo "Hello, Earth!"
---
Hello, Earth!
---
✅ OK

echo "Hello, $CAPTURED_WORD!"; echo "Goodbye, Earth!"
---
Hello, world!
Goodbye, Earth!
---
✅ OK

<root>/capture.cli PASSED